### PR TITLE
Report the error more gracefully when nvidia-smi not exist

### PR DESCRIPTION
**Pull request by liuzhe-lz**
Currently `nni_gpu_tool` crashes when `nvidia-smi` is not available.
This PR makes it output empty GPU status (`"gpuCount": 0`) instead.
In local mode, when `gpuNum` set to non-zero and `gpuCount` is zero, the experiment fail.
In remote mode, when a server's `gpuCount` is zero, a **warning** message will be written to log file. (If you think it should be an "error", please leave comment.)

Update:
Previously the script creates `gpu_metrics` file and then changes it's permission to `777`. This will fail when the file is owned by another user, even if it's `777`.
Now it creates `gpu_metrics` with `umask` 0.

Relate to: issue #1375

### CODE REVIEW

File: tools/nni_gpu_tool/gpu_metrics_collector.py [hunk 1]
```diff
@@ -48,11 +49,13 @@ def main(argv):
     with open(os.path.join(metrics_output_dir, "gpu_metrics"), "w") as outputFile:
         pass
     os.chmod(os.path.join(metrics_output_dir, "gpu_metrics"), 0o777)
-    cmd = 'nvidia-smi -q -x'
+    cmd = 'nvidia-smi -q -x'.split()
     while(True):
         try:
-            smi_output = subprocess.check_output(cmd, shell=True)
+            smi_output = subprocess.check_output(cmd)
             parse_nvidia_smi_result(smi_output, metrics_output_dir)
```
Comment (QuanluZhang) [inline tools/nni_gpu_tool/gpu_metrics_collector.py]: there is try catch within `parse_nvidia_smi_result`

Comment (QuanluZhang) [reply tools/nni_gpu_tool/gpu_metrics_collector.py]: and it uses 'a' in `open`

Comment (liuzhe-lz) [reply tools/nni_gpu_tool/gpu_metrics_collector.py]: This is to catch `subprocess.check_output` when `nvidia-smi` not exist

File: src/nni_manager/training_service/local/gpuScheduler.ts [hunk 2]
```diff
@@ -54,6 +54,9 @@ class GPUScheduler {
             } catch (error) {
                 this.log.error('Read GPU summary failed with error: ', error);
             }
+            if (this.gpuSummary.gpuCount === 0) {
+                throw new Error('GPU not available. Please check your CUDA configuration');
```
Comment (SparkSnail) [inline src/nni_manager/training_service/local/gpuScheduler.ts]: If the gpu_metrics_collector update gpu_metric file slowly, will gpuSummary.gpuCount updated to a positive number after several detection?

**Pull request merged**